dubbo-rpc/dubbo-rpc-dubbo/src/main/java/org/apache/dubbo/rpc/protocol/dubbo/telnet/ListTelnetHandler.java

```diff
@@ -54,47 +60,95 @@ public String telnet(Channel channel, String message) {
         } else {
             service = (String) channel.getAttribute(ChangeTelnetHandler.SERVICE_KEY);
             if (service != null && service.length() > 0) {
-                buf.append("Use default service " + service + ".\r\n");
+                buf.append("Use default service ").append(service).append(".\r\n");
             }
         }
-        if (service == null || service.length() == 0) {
-            for (Exporter<?> exporter : DubboProtocol.getDubboProtocol().getExporters()) {
-                if (buf.length() > 0) {
-                    buf.append("\r\n");
-                }
-                buf.append(exporter.getInvoker().getInterface().getName());
-                if (detail) {
-                    buf.append(" -> ");
-                    buf.append(exporter.getInvoker().getUrl());
-                }
-            }
+
+        if (StringUtils.isEmpty(service)) {
+            printAllServices(buf, detail);
         } else {
-            Invoker<?> invoker = null;
-            for (Exporter<?> exporter : DubboProtocol.getDubboProtocol().getExporters()) {
-                if (service.equals(exporter.getInvoker().getInterface().getSimpleName())
-                        || service.equals(exporter.getInvoker().getInterface().getName())
-                        || service.equals(exporter.getInvoker().getUrl().getPath())) {
```

Comment:
`service.equals(exporter.getInvoker().getUrl().getPath())` 
This condition will lost in the new implementation while the Model concept used only takes `group/service:version` into account. I am a little confused about the relation between `path` and `group/service:version` at this moment, but I think we should be careful before we can think of them clearly enough.

Reply:
Yes agree that for changes we should be thought full. In this case what approach we could take to make it better. If you could guide, we would make change to enhance it?